src/main/java/com/basilisk/frontend/components/TweetDisplayComponent.java

```diff
@@ -86,6 +89,7 @@ public TweetDisplayComponent(TweetPresenter tweetPresenter, Tweet tweet) {
     public TweetDisplayComponent(TweetPresenter tweetPresenter, Retweet retweet) {
         this.tweetPresenter = tweetPresenter;
         this.tweet = retweet.getTweet();
+        this.retweet = retweet;
```

Comment:
Interesting, I guess its true that you need the retweet object if you wanna be able to delete it.